### DISCUSSION TITLE
Patching store returns compile error in vscode

### DISCUSSION THREAD

**Discussion by Rednas83** [vuejs/pinia#2527] [Help and Questions] [answered]
### Reproduction

Not applicable

### Steps to reproduce the bug

**useUserStore.ts**
```
export const useUserStore = defineStore("user", () => {
  const name = ref("pinia")
  const level = ref(1)
  const debug = ref(true)
  const diagnostics = ref(false)
  return { name, level, debug, diagnostics }
})
```

**app.vue**
```
const user = useUserStore()
user.$state = { name: popup.identifier, level: 1, debug: false, diagnostics: false }
```
OR
```
const user = useUserStore()
user.$patch({ name: popup.identifier, level: 1, debug: false, diagnostics: false })
```

### Expected behavior

No compile errors. Was there a breaking change? This code worked before

### Actual behavior

**$state**
![image](https://github.com/vuejs/pinia/assets/25219229/2d34911c-6c41-49b6-8dc5-57b3bb0a7b19)
**$patch**
![image](https://github.com/vuejs/pinia/assets/25219229/fac0bf66-803e-4bb3-acad-c070593e083b)

### Additional information

**package.json**
devDependencies:{
"@antfu/eslint-config": "2.4.5",
"eslint": "^8.55.0",
"nuxt": "^3.8.2",
}
"dependencies": {
"@pinia/nuxt": "^0.5.1",
"pinia": "^2.1.7",
}

**Comment by posva** [comment, marked as answer]
It seems to work: https://play.pinia.esm.dev/#eNqNVE2P2jAQ/StWVIlFJXbptnuIYEVb7aE9tFW3R19CMgRTf8l2WCrEf+/ECSHQ3RW3zMzzzJvh8fbJJ2vptoYkS2a+cMIG4iHUlshcV3OeBM+Te66FssYFsie1h8dgHJADWTmjyIiywtQ6gKPBj7jmujDaY5MImvf4mzHHNH7QNz7koSntic4VZGS0BinNaEIkbEFmZDohJSzrKiOrXHrASOSVNj6Iwnc5cuB6xlq+yA6DAMpK7IvRjPVBMkla5qnKLd14o3HPPdeE8K6A62UkZpocHqKJebIOwfqMsaLU+KwEKbaOaghMW8UWCGMOlxYK0tKoxS29pdP3rBQ+DPMUvEqXzjx5vM4GJ00Gcxgmt+BSB7oEB+7auRfPzmZf1F6eb4UW+flEPNhfGvPNM1rClvXR800iHcQFY6RPcyuuXWH4ZnFHP9Ipk2LJcBATyHz3363SEtQ13Y/QxTs6/UDv2FtsyZOmEerlgGIIHtW5EtWFFAqjrJDgftggUL1nkshRm0/fYi64GnpaxRqKP8/kN37XMv3pIP4eg1VC7ioIbfnh8Tvs8LsvKlPWspPfC8Vf4I2sG44t7HOtS6Q9wEW2X6Oyha5++4ddAO2PSzVE4zUiPp72yyurn+iiyAZXPP3h8YS9M5SwEvrCHKKC0BV6kINVX8Tp0TBgF4utb/T+Mh82vOEJFppFyc2YzO9bju2LxkQQjZ0R1QkbzeZYjqbS1aeDfLSYLt/cZVg6+U0HiKYTEQ7N0enOuzrL6vzqzKjircbJ4R8JrNJ/
